pkg/kubelet/kubelet_pods.go

```diff
@@ -135,7 +136,37 @@ func makeMounts(pod *v1.Pod, podDir string, container *v1.Container, hostName, h
 			return nil, err
 		}
 		if mount.SubPath != "" {
+			fileinfo, err := os.Lstat(hostPath)
+			if err != nil {
+				return nil, err
+			}
+			gid := fileinfo.Sys().(*syscall.Stat_t).Gid
+			perm := fileinfo.Mode()
+
 			hostPath = filepath.Join(hostPath, mount.SubPath)
+
+			// If fsGroup volume ownership was set for the mounter path, set it for the sub path too
+			fsGroup := pod.Spec.SecurityContext.FSGroup
+			if gid == uint32(*fsGroup) {
+				if err := os.MkdirAll(hostPath, perm); err != nil {
+					glog.Errorf("failed to mkdir:%s", hostPath)
+					return nil, err
+				}
+				volume.SetVolumeOwnership(vol.Mounter, fsGroup, hostPath)
+
+				// If umask is preventing us from making the sub path have the same permissions as the
+				// mounter path, chmod the sub path
+				fileinfo, err := os.Lstat(hostPath)
```

Comment:
I don't understand.

`SetVolumeOwnership()` ORs with 0660, and then calls `Chmod()` - the same as you are here.  Umask might apply in the mkdir, but as far as I can tell your Chmod is the same as the one called deep below SetVolumeOwnership